From da83d31a8dd555c28eb27bd7a7d3c53b51d5f276 Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Sun, 20 Jul 2025 16:37:57 +0800
Subject: [PATCH] eventfd.cpp: Remove the scope resolution operator

Remove the scope resolution operator :: to fix the below build failure
with musl.
 | ../../../sources/lttng-tools-2.14.0/src/common/eventfd.cpp:18:31: error: expected id-expression before numeric constant
 |    18 |                 int flags = ::EFD_CLOEXEC;
 |       |                               ^~~~~~~~~~~
 | ../../../sources/lttng-tools-2.14.0/src/common/eventfd.cpp:21:36: error: expected id-expression before numeric constant
 |    21 |                         flags |= ::EFD_SEMAPHORE;
 |       |                                    ^~~~~~~~~~~~~

Upstream-Status: Pending [The message to lttng-dev awaits moderator approval]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 src/common/eventfd.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/common/eventfd.cpp b/src/common/eventfd.cpp
index fe5b549..3379b0b 100644
--- a/src/common/eventfd.cpp
+++ b/src/common/eventfd.cpp
@@ -15,10 +15,10 @@
 
 lttng::eventfd::eventfd(bool use_semaphore_semantics, std::uint64_t initial_value) :
 	file_descriptor([use_semaphore_semantics, initial_value]() {
-		int flags = ::EFD_CLOEXEC;
+		int flags = EFD_CLOEXEC;
 
 		if (use_semaphore_semantics) {
-			flags |= ::EFD_SEMAPHORE;
+			flags |= EFD_SEMAPHORE;
 		}
 
 		const auto raw_fd = ::eventfd(initial_value, flags);
-- 
2.34.1

